#209 FICS premove

ukscid opened the pull request:
Premove (as a illegal move) in current position was  ignored by addMove. Looking for fics::premove in addMoveUCI is "to late"

Commit:
Bugfix Premove

Commit:
revert change for reviewgame::playerCanMove

benini commented:
This doesn't seems ok: the direct invocation of addMoveUCI will now bypass FICS's checks.
Nonetheless, these checks should be within the FICS module rather than within this functions.

ukscid commented:
I can't see where addMoveUCI is called direct for doing a premove.
But adding the fics's check in addMoveUCI again is also not the full fix.
I see that a premove with a promotion throws an error.
I'll investigate.

Commit:
Bugfix premove

premove was executed every time a string receives from fics, but fics send empty string after a move. This triggered a second premove during promotion request. Now premove will only executed after a move is received from fics.
